message_ix/tests/test_feature_temporal_level.py

```diff
@@ -165,173 +37,123 @@ def model_generator(
-    """
-    Testing two unlinked temporal levels.
+    for h in act.loc[act["technology"] == "gas_ppl"]["time"]:
+        act.loc[
+            (act["time"] == h) & (act["technology"] == "gas_ppl"),
+            "capacity-corrected",
+        ] = act["lvl"] / float(scen.par("duration_time", {"time": h})["value"])
+    assert max(act.loc[act["technology"] == "gas_ppl", "capacity-corrected"]) == float(
+        scen.var("CAP", {"technology": "gas_ppl"})["lvl"]
+    )
+
+
+# Dictionary of technology input/output
+TD_0 = {
+    "gas_ppl": {
+        "time_origin": [],
+        "time": ["summer"],
+        "time_dest": ["summer"],
+    },
+    "gas_supply": {"time_origin": [], "time": ["year"], "time_dest": ["year"]},
+}
+
 
-    "gas_ppl" is active in "summer" and NOT linked to "gas_supply" in "year"
-    This setup should not solve.
+def test_temporal_levels_not_linked(request):
+    """Test two unlinked temporal levels.
 
-    Parameters
-    ----------
-    test_mp : ixmp.Platform()
+    "gas_ppl" is active in "summer" and NOT linked to "gas_supply" in "year". This
+    setup should not solve.
     """
-    comment = "1.not-linked"
-    # Dictionary of technology input/output
-    tec_dict = {
-        "gas_ppl": {
-            "time_origin": ["summer"],
-            "time": ["summer"],
-            "time_dest": ["summer"],
-        },
-        "gas_supply": {"time_origin": [], "time": ["year"], "time_dest": ["year"]},
-    }
+    tec_dict = TD_0.copy()
+    tec_dict["gas_ppl"]["time_origin"] = ["summer"]
 
     # Check the model not solve if there is no link between fuel supply and power plant
     with pytest.raises(ModelError):
-        model_generator(
-            test_mp,
-            comment,
+        make_subannual(
+            request,
             tec_dict,
             time_steps=[("summer", 1, "season", "year")],
             demand={"summer": 2},
+            com_dict=COM_DICT,
         )
 
 
-def test_season_to_year(test_mp):
-    """
-    Testing two linked temporal levels.
+def test_season_to_year(request):
+    """Test two linked temporal levels.
 
     Linking "gas_ppl" and "gas_supply" at one temporal level (e.g., "year")
     Only one "season" (duration = 1) and "demand" is defined only at "summer"
     Model solves and "gas_supply" is active.
-
-    Parameters
-    ----------
-    test_mp : ixmp.Platform()
     """
-    comment = "2.linked-one-season-with-year"
-    # Dictionary of technology input/output
-    tec_dict = {
-        "gas_ppl": {
-            "time_origin": ["year"],
-            "time": ["summer"],
-            "time_dest": ["summer"],
-        },
-        "gas_supply": {"time_origin": [], "time": ["year"], "time_dest": ["year"]},
-    }
-    model_generator(
-        test_mp,
-        comment,
+    tec_dict = TD_0.copy()
+    tec_dict["gas_ppl"]["time_origin"] = ["year"]
+
+    scen = make_subannual(
+        request,
         tec_dict,
         time_steps=[("summer", 1, "season", "year")],
```

Comment:
We could also think about moving this up and create e.g. TS_0 to reduce duplication (this line and line 75).

Reply:
Ah, true. I skipped this one because it doesn't save on # of lines.